documentation/developers/mapping_instructions.md

```diff
@@ -0,0 +1,748 @@
+    [
+        {
+            "id": "2aff48d8-917a-4c9b-814d-99639a220773",
+            "address": "74 Whyioughta St."
+        },
+        {
+            "id": "a4354016-1a5a-47ef-9d14-01217f33ff2e",
+            "address": "1600 Ipoke Ave"
+        }
+    ]
+    ```
+    So, using #!entity-reference, how can we make sure everyone is assigned to the proper house? An anchor reference.
+    The first argument should be the path within the source that describes the output entity, in this case a Person. A path that best represents a person in the source is `households[*].names[*]`. The second argument decribes the source of the information in the source document, in this case it's a House, so the proper path would be `households[*]`. As an full command, we get:
+    
+    ```
+    src:    #!entity-reference#House[*].id#households[*].names[*]#households[*]
+    dst:    Person.house
+    ```
+    The new values of the Stooges will have them properly referenced to their respective houses:
+    ```json
+    [
+        {
+            "id": "a7a1f889-9223-44a2-bbea-b41a415c2989",
+            "name": "Larry",
+            "number": 1,
+            "house": "2aff48d8-917a-4c9b-814d-99639a220773"
+        },
+        {
+            "id": "40188061-1942-4041-ae1c-1e1d99221698",
+            "name": "Curly",
+            "number": 2,
+            "house": "2aff48d8-917a-4c9b-814d-99639a220773"
+        },
+        {
+            "id": "767f38e8-ce3c-41ee-bd4f-cc1c41bbbde9",
+            "name": "Moe",
+            "number": 3,
+            "house": "a4354016-1a5a-47ef-9d14-01217f33ff2e"
+        }
+    ]
+    ```
+
+    Notice that Larry and Curly still live together, and Moe still lives alone.
+
+
+
+
+---
+
+<a name="extractor-mechanism"></a>
+## Extractor Mechanism
+
+As stated previously, Mapping instruction sets are executed in order. If the extractor encounters an error on an instruction, it caches that instuction it and continues down the list. Once finished with the list, it will reattempt the instuctions that had errors. It will continue to retry until either all instuctions have been completed _or_ a round occurs where none of the cached erroneous instuctions were able to be completed. If there are errors at the end of extraction, the errors are reported.
+
+Let's introduce some new artifacts so we can illustrate the process from the extractor point of view.
+
+<details open>
+    <summary><i>Input <b>(Hide/Show)</b></i></summary>
+    
+```json
+{
+    "family_name": "Keaton",
+    "address":"1000 Chortle Way",
+    "adults": [
+        {"name": "Buster"},
+        {"name": "Grace"}
+    ],
+    "children" : [
+        {"name": "Ace"},
+        {"name": "Franklin"}
+    ]
+}
+```
+
+</details>
+<details open>
+    <summary><i>Schemas <b>(Hide/Show)</b></i></summary>
+
+```json
+[
+  {
+    "name": "Person",
+    "type": "record",
+    "fields": [
+      {
+        "name": "id",
+        "type": "string",
+        "jsonldPredicate": "@id"
```

Comment:
Is this (`jsonldPredicate`) relevant for this topic?

Reply:
It's going to be a whole other topic, but I want to consistently include it so it doesn't seem optional as one day it will be used heavily.